frontend/extensions/helpers.lua

```diff
@@ -0,0 +1,275 @@
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local Screen = require("device").screen
+
+local SQ3 = require("lua-ljsqlite3/init")
+local DataStorage = require("datastorage")
+local LuaSettings = require("luasettings")
+local UIManager = require("ui/uimanager")
+local InfoMessage = require("ui/widget/infomessage")
+local Font = require("ui/font")
+local ConfirmBox = require("ui/widget/confirmbox")
+local _ = require("gettext")
+local lfs = require("libs/libkoreader-lfs")
+
+-- Plugin for global methods
+-- Some modules don't use this, but communicate directly with each other by exchanging global variables by triggering events. For an example, search for variable AX_remaining_time.
+
+-- for widgets use :extend instead of :new ?:
+local Helpers = WidgetContainer:extend{
+
+	debug_once_used = false,
+
+	file_paths = {},
+
+	garbage = nil,
+}
+
+function Helpers:init(ui)
+	self.ui = ui
+end
+
+function Helpers:getHomeDir()
+
+	local homedir = G_reader_settings:readSetting("home_dir")
+
+	local current_ebook
+
+	if homedir == nil then
+		current_ebook = G_reader_settings:readSetting("lastfile")
+
+		if current_ebook ~= nil then
+			homedir = string.gsub(current_ebook, '/[- .,:;?!_A-Za-z0-9]+', '')
+		end
+    end
+
+	return homedir
+end
+
+function Helpers:getKoreaderDir()
```

Comment:
I agree with @poire-z that there's a lot of stuff in this file that doesn't really seem to fit. This one for example is doing the reverse of `DataStorage:getSettingsDir()`, which is just `DataStorage:getDataDir()` + `/settings`/ I believe?

Reply:
Maybe you're right. I'll look it up. I've learned lua by looking at the KOReader source code, so my code will not be always optimal.